packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -725,7 +725,9 @@ export class BasicCrawler<
 
         try {
             await addTimeoutToPromise(
-                () => this._runRequestHandler(crawlingContext),
+                crawlingContext.request.skipNavigation
+                    ? (() => this.requestHandler(crawlingContext) as any)
+                    : () => this._runRequestHandler(crawlingContext),
```

Comment:
I don't think this is correct. There are many actions that are happening in the `_runRequestHandler` functions for all the crawlers except `Basic`. All those actions need to complete. The only thing we should skip is the navigation itself = the HTTP request in Cheerio and the `page.goto` call in browsers.

For `Basic`, `skipNavigation` probably does not make sense at all, because the navigation is never handled by the crawler.

Reply:
This follows the @B4nan 's [idea](https://github.com/apify/apify-ts/issues/79#issuecomment-1110237785) from the original issue (which I like as makes sense in the OOP inheritance way, even though `BasicCrawler` itself does not navigate). 
Although I do see that the `skipNavigation` naming suggests only skipping _navigation_, it seems to me that with Cheerio/BrowserCrawler, the rest of the actions are pretty much no-op without the navigation taking place. 

I'm sure you know better though, seeing a counterexample / reason why would the users want it would be cool, if possible :)

Reply:
I don't see a single no-op here in `BrowserCrawler` :) The user wouldn't even have a `page` to work with. With Cheerio most do seem like a no-op, that's true. But still not all of them. E.g. the `proxyInfo` should be set, so that the user can make their own navigation through the proper proxy.

https://github.com/apify/apify-ts/blob/c7f7b13a0ca3df9595b37aa5756fbd34b6de4502/packages/browser-crawler/src/internals/browser-crawler.ts#L472-L500

Reply:
Well, given the original issue name ([Handle static requests outside of the browser's page](https://github.com/apify/apify-ts/issues/79)), I was under the impression that the user would not need the... page? :) 
But it's definitely your call, will update and we can discuss that over the updated version.

Reply:
Yeah, the original idea was to enable making HTTP requests from browser crawlers, but that's just one use-case out of many. And instead of supporting this one case specifically, we decided to go with a generic solution that will skip navigation and leave everything else to the user. But you're right that for the HTTP request use-case the page does not make sense.